inc/Engine/Optimization/RUCSS/Warmup/WarmupProcess.php

```diff
@@ -0,0 +1,105 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Engine\Optimization\RUCSS\Database\Queries\ResourcesQuery;
+use WP_Rocket\Engine\Optimization\RUCSS\Database\Row\ResourceRow;
+use \WP_Rocket_WP_Background_Process;
+
+class WarmupProcess extends WP_Rocket_WP_Background_Process {
+
+	/**
+	 * Background process action name.
+	 *
+	 * @var string
+	 */
+	protected $action = 'rucss_warmup_saas_call';
+
+	/**
+	 * Background process prefix.
+	 *
+	 * @var string
+	 */
+
+	protected $prefix = 'rocket';
+
+	/**
+	 * ResourcesQuery instance.
+	 *
+	 * @var ResourcesQuery
+	 */
+	private $resources_query;
+
+	/**
+	 * APIClient instance.
+	 *
+	 * @var APIClient
+	 */
+	private $api_client;
+
+	/**
+	 * WarmupProcess constructor.
+	 *
+	 * @param ResourcesQuery $resources_query ResourcesQuery instance.
+	 * @param APIClient      $api_client      APIClient instance.
+	 */
+	public function __construct( ResourcesQuery $resources_query, APIClient $api_client ) {
+		parent::__construct();
+
+		$this->resources_query = $resources_query;
+		$this->api_client      = $api_client;
+	}
+
+	/**
+	 * Background process task for each resource
+	 *
+	 * @param int $resource_id Resource DB id.
+	 *
+	 * @return bool
+	 */
+	protected function task( $resource_id ) {
+		if ( ! is_int( $resource_id ) ) {
+			return false;
+		}
+
+		$resource_row = $this->resources_query->get_item( $resource_id );
```

Comment:
We already have the information we need from the previous background process, why do we do a new request in the database for them?

Can't we re-use directly the initial data, by passing everything to the `$send_to_warmup` array?

Reply:
really we don't have the DB row in all cases as this method 

https://github.com/wp-media/wp-rocket/blob/d7fdf63f9e5249dfe0ae6305b4a7b2b854129368/inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcherProcess.php#L65

returns the row ID only so in this case we would need to make another DB query to get this row.

also this update method

https://github.com/wp-media/wp-rocket/blob/d7fdf63f9e5249dfe0ae6305b4a7b2b854129368/inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcherProcess.php#L97

returns only ID, so if U used `$db_row` it wouldn't be updated.

for those two reasons I took the decision to pass only the IDs and there I can query to get them

Reply:
This is the flow I currently see:
- In `ResourceFetcher`, the `resources` property array contains the URL, content and type
- They are passed to the `ResourceFetcherProcess` to be saved in the DB, and return the ID
- In `WarmupProcess`, we get the same 3 values from the DB, by fetching them from the DB by the ID previously created

Do you see the issue here? At the start we have the data we need for the end, but instead of passing that data all the way, we add an additional step of reading again from the DB to get the same data.

Reply:
Got u, so u need me to pass the data directly to the warmup process, good idea!
Will adjust that now.

Reply:
It does make more sense to do it that way, don't you think? It avoids an extra request to the DB.

Reply:
yes but when will we do the following DB update?
https://github.com/wp-media/wp-rocket/blob/504c70adbea734d97e7f4c73c678448b103d54fa/inc/Engine/Optimization/RUCSS/Warmup/WarmupProcess.php#L93-L98

really I don't think that we need this column `resend_to_warmup` anymore as we already pass the resource itself not making a query to get all resources need to be sent to warmup.

Or will we make cleanup cron for those not warmedup resources for any reason? => I don't think so.

what do u think ?

Reply:
I was wondering that myself earlier. I think that unless we need it for some other reason, (will we ever need to preserve that info in case of a retry?) we can probably remove it now.